base/runtime/random_generator.odin

```diff
@@ -86,16 +86,23 @@ default_random_generator_proc :: proc(data: rawptr, mode: Random_Generator_Mode,
 		   	init(r, 0)
 		}
 
-		pos := i8(0)
-		val := u64(0)
-		for &v in p {
-			if pos == 0 {
-				val = read_u64(r)
-				pos = 7
+		switch len(p) {
+		case size_of(u64):
+			// Fast path for a 64-bit destination.
+			(transmute(^u64)raw_data(p))^ = read_u64(r)
```

Comment:
Maybe replace it with `intrinsics.unaligned_store` and call it a day?

Reply:
I'm unfamiliar with this situation. What happens if the pointer isn't aligned? Does it write as if it was aligned and overwrite nearby memory? Is this dangerous only on certain platforms or can I pass a non-aligned pointer in myself to see the effects?

I'm guessing this isn't something that can be guarded against with a `when #align_of` construction, is it? Because it's the pointer, and not the pointer type, that we're concerned about for alignment.

It's not a big deal to me if this has to be reverted. I was just trying out writing my own RNG and noticed this as a possible optimization.

Reply:
It's fixed by my subsequent commit. But basically it's either illegal or really, really slow on some platforms to perform unaligned reads and/or writes. And if you don't know if it is, you may want to use the intrinsic will does a quick alignment check and then either does the pointer write or a memcpy.

Reply:
Aha. I was aware of unaligned reads/writes being slow, but not that they are illegal on some platforms. Good to know about `unaligned_store`.